### DISCUSSION TITLE
How to identify the type of model or monai label app currently running on the server

### DISCUSSION THREAD

**Discussion by saileshsidhwani** [Project-MONAI/MONAILabel#1356] [Q&A]
If I understand correctly, identifying the type of models is generally straightforward. This refers to whether the model is for segmentation, detection, classification, or another type. This information is typically available by accessing the 'info' of the app running on the server via http://127.0.0.1:8000/info/.

What I was wondering was how a client would identify more information about the app currently running on the server.

1. What type of app is running? (radiology/pathology/endoscopy). Theoretically, it is possible for a single app to contain models/bundles from multiple domains. I know it's not common, but I think it is possible. How would a client guard against this?

2. If question number 1 is not directly answerable, another way to guard against this would be to filter out the models inside the app along the lines of model type, model inputs, model outputs, model dimensions, etc. For example, the type of model in the info json can help decide the InferTask type. But that alone isn't enough when say a client is only interested in supporting just one of the domains and the info json does not include information line model inputs/output dimension.

**Comment by saileshsidhwani** [comment]
Bubbling this up to get some fresh eyes on this

**Reply by SachidanandAlle** [reply]
We can add additional information for each model.. an /info/ can receive the same..

**Comment by tangy5** [comment]
Thanks for the question. Technically, we can add more information through /info, and identify which app, image modality, or domain it belongs. Instead of a specific app, would a description be better? Both case, it's possible, you could add this to the /info/. 
Thanks
